src/pool/pool_disjoint.c

```diff
@@ -804,11 +821,21 @@ umf_result_t disjoint_pool_free(void *pool, void *ptr) {
 
     bucket_t *bucket = slab->bucket;
 
-    VALGRIND_DO_MEMPOOL_FREE(pool, ptr);
     utils_mutex_lock(&bucket->bucket_lock);
 
-    utils_annotate_memory_inaccessible(ptr, bucket->size);
-    bucket_free_chunk(bucket, ptr, slab, &to_pool);
+    // TODO valgrind
```

Comment:
Generally, I don't care about comments, but in one commit there are such styles:
`TODO sth`
`TODO - sth`
`NOTE: std`

Reply:
done